File: src/jk_5/nailed/command/CommandTeamleader.java
package jk_5.nailed.command;

import jk_5.nailed.Nailed;
import jk_5.nailed.map.Map;
import jk_5.nailed.teams.Team;
import jk_5.nailed.players.Player;
import jk_5.nailed.util.EnumColor;
import net.minecraft.server.MinecraftServer;
import net.minecraft.src.*;

import java.util.List;

/**
 * No description given
 *
 * @author jk-5
 */
public class CommandTeamleader extends CommandBase {

    @Override
    public String getCommandName() {
        return "teamleader";
    }

    @Override
    public int getRequiredPermissionLevel() {
        return 2;
    }

    @Override
    public String getCommandUsage(ICommandSender par1ICommandSender) {
        return "/teamleader - Sets the teamleaders";
    }

    @Override
    public void processCommand(ICommandSender send, String[] args) {
        if (!(send instanceof EntityPlayerMP)) return;
        EntityPlayerMP sender = (EntityPlayerMP) send;
        Map currentMap = Nailed.mapLoader.getMapFromWorld(sender.getServerForPlayer());
        if (args.length == 0) {
            throw new WrongUsageException("/teamleader <set:unset> <player>");
        }else if(args.length == 1){
            if(args[0].equals("unset") || args[0].equals("set")){
                throw new WrongUsageException("/teamleader " + args[0] + " <player>");
            }
        }else if(args.length == 2){
            if(args[0].equals("set")){
                Player player = Nailed.playerRegistry.getPlayer(args[1]);
                if(player == null) throw new CommandException("Player " + args[1] + " was not found");
                if(player.getTeam() == Team.UNKNOWN) throw new CommandException("Player " + args[1] + " does not have a team");
                player.getTeam().setTeamLeader(player);
                currentMap.sendMessageToAllPlayersWithoutPrefix(player.getGroup().getChatPrefix() + player.getChatFormattedName() + EnumColor.GREEN + " is now the leader of team " + player.getTeam().toString());
            }else if(args[0].equals("unset")){
                Player player = Nailed.playerRegistry.getPlayer(args[1]);
                if(player == null) throw new CommandException("Player " + args[1] + " was not found");
                if(player.getTeam() == Team.UNKNOWN) throw new CommandException("Player " + args[1] + " does not have a team");
                if(player.getTeam().getTeamLeader() != player) throw new CommandException("Player " + args[1] + " is not the teamleader of his team");
                player.getTeam().setTeamLeader(null);
                currentMap.sendMessageToAllPlayersWithoutPrefix(player.getGroup().getChatPrefix() + player.getChatFormattedName() + EnumColor.GOLD + " is no longer the leader of team " + player.getTeam().toString());
            }
        }else throw new WrongUsageException("/teamleader <set:unset> <player>");
    }

    @Override
    public List addTabCompletionOptions(ICommandSender send, String[] args) {
        if (!(send instanceof EntityPlayerMP)) return null;
        EntityPlayerMP sender = (EntityPlayerMP) send;
        if (args.length == 1) {
            return getListOfStringsMatchingLastWord(args, "set", "unset");
        } else if (args.length == 2) {
            if (args[0].equals("set") || args[0].equals("unset")) {
                Player p = Nailed.playerRegistry.getPlayer(args[1]);
                if(p == null) return null;
                return getListOfStringsFromIterableMatchingLastWord(args, p.getTeam().getAllPlayerNames());
            }
        }
        return null;
    }
}

File: src/main/scala/jk_5/nailed/command/CommandTeamleader.scala
package jk_5.nailed.command

import jk_5.nailed.players.PlayerRegistry
import jk_5.nailed.team.Team
import jk_5.nailed.util.EnumColor
import scala.collection.mutable

/**
 * No description given
 *
 * @author jk-5
 */
object CommandTeamleader extends TCommand {
  val commandName = "teamleader"
  this.permissionLevel = 2

  override def getCommandUsage = "/teamleader - Sets the teamleaders"
  def processCommand(sender: CommandSender, args: Array[String]){
    val currentMap = sender.map
    args.length match {
      case 0 => throw new WrongUsageException("/teamleader <set:unset> <player>")
      case 1 => if(args(0) == "unset" || args(0) == "set") throw new WrongUsageException("/teamleader <set:unset> <player>")
      case 2 => {
        if(args(0) == "set"){
          val player = PlayerRegistry.getPlayer(args(1))
          if(player.isEmpty) throw new CommandException("Player " + args(0) + " was not found")
          if(player.get.getTeam == Team.UNKNOWN) throw new CommandException("Player " + args(1) + " does not have a team")
          player.get.getTeam.setTeamLeader(player.get)
          currentMap.sendMessageToAllPlayersWithoutPrefix(player.get.getGroup.getChatPrefix + player.get.getChatFormattedName + EnumColor.GREEN + " is now the leader of team " + player.get.getTeam)
        }else if(args(0) == "unset"){
          val player = PlayerRegistry.getPlayer(args(1))
          if(player.isEmpty) throw new CommandException("Player " + args(0) + " was not found")
          if(player.get.getTeam.getTeamLeader == player.get) throw new CommandException("Player " + args(1) + " is not the teamleader of his team")
          player.get.getTeam.setTeamLeader(null)
          currentMap.sendMessageToAllPlayersWithoutPrefix(player.get.getGroup.getChatPrefix + player.get.getChatFormattedName + EnumColor.GOLD + " is no longer the leader of team " + player.get.getTeam)
        }
      }
      case _ => throw new WrongUsageException("/teamleader <set:unset> <player>")
    }
  }

  override def addAutocomplete(s: CommandSender, args: Array[String]): mutable.Buffer[String] = args.length match{
    case 1 => Command.getMatched(args, "set", "unset")
    case 2 => if(args(0) == "set" || args(0) == "unset"){
      val p = PlayerRegistry.getPlayer(args(1))
      if(p.isEmpty) return null
      Command.getMatched(args, p.get.getTeam.getAllPlayerNames)
    } else null
  }
}
